### ISSUE TITLE
Release a new version of the VS Code extension to the marketplace

### ISSUE THREAD

**Issue by jonasbb** [bash-lsp/bash-language-server#328]
The extension on the [VS Code marketplace](https://marketplace.visualstudio.com/items?itemName=mads-hartmann.bash-ide-vscode) is still at version 1.11.0 and ships with the language server in version 1.16.1. The last update happened on 2020-08-24.
Based on the code in https://github.com/bash-lsp/bash-language-server/tree/master/vscode-client it looks like there should be a newer version 1.12.0 which includes the language server in version 2.0.0. The changelog already mentions version 1.12.0, but the `package.json` is still at 1.11.0.

**Comment by skovhus** [comment]
@jonasbb new version is out now. Thanks for raising this. Does it work for you?

**Comment by jonasbb** [comment]
@skovhus Hi, thanks for releasing a new version. Unfortunately, it crashes directly after starting. Here is the log from the output console of VS Code. I am using VS Code 1.64.2 on Fedora 35.

```text
internal/modules/cjs/loader.js:937
  throw err;
  ^

Error: Cannot find module 'side-channel'
Require stack:
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/index.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/lib/querystring.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/request.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/index.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request-promise-native/lib/rp.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/jsdom/lib/api.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/turndown/lib/turndown.cjs.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/bash-language-server/out/server.js
- /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/out/server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:934:15)
    at Module._load (internal/modules/cjs/loader.js:779:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
    at Module.require (internal/modules/cjs/loader.js:1006:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js:3:22)
    at Module._compile (internal/modules/cjs/loader.js:1125:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
    at Module.load (internal/modules/cjs/loader.js:982:32)
    at Module._load (internal/modules/cjs/loader.js:823:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/index.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/lib/querystring.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/request.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/index.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request-promise-native/lib/rp.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/jsdom/lib/api.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/turndown/lib/turndown.cjs.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/bash-language-server/out/server.js',
    '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/out/server.js'
  ]
}
[Info  - 3:04:06 PM] Connection to server got closed. Server will restart.
```

The message repeats a couple of times until it ends with
```text
[Error - 3:04:08 PM] Connection to server got closed. Server will not be restarted.
```

The other day I used the GitHub Codespaces to compile myself a vsix from the master branch. That worked fine and I can use for example the bash-completion support with that.

**Comment by AgentOfTech** [comment]
> @skovhus Hi, thanks for releasing a new version. Unfortunately, it crashes directly after starting. Here is the log from the output console of VS Code. I am using VS Code 1.64.2 on Fedora 35.
> 
> ```
> internal/modules/cjs/loader.js:937
>   throw err;
>   ^
> 
> Error: Cannot find module 'side-channel'
> Require stack:
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/index.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/lib/querystring.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/request.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/index.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request-promise-native/lib/rp.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/jsdom/lib/api.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/turndown/lib/turndown.cjs.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/bash-language-server/out/server.js
> - /home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/out/server.js
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:934:15)
>     at Module._load (internal/modules/cjs/loader.js:779:27)
>     at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
>     at Module.require (internal/modules/cjs/loader.js:1006:19)
>     at require (internal/modules/cjs/helpers.js:88:18)
>     at Object.<anonymous> (/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js:3:22)
>     at Module._compile (internal/modules/cjs/loader.js:1125:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1155:10)
>     at Module.load (internal/modules/cjs/loader.js:982:32)
>     at Module._load (internal/modules/cjs/loader.js:823:14) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/stringify.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/qs/lib/index.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/lib/querystring.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/request.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request/index.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/request-promise-native/lib/rp.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/jsdom/lib/api.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/turndown/lib/turndown.cjs.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/node_modules/bash-language-server/out/server.js',
>     '/home/jbushart/.vscode/extensions/mads-hartmann.bash-ide-vscode-1.12.0/out/server.js'
>   ]
> }
> [Info  - 3:04:06 PM] Connection to server got closed. Server will restart.
> ```
> 
> The message repeats a couple of times until it ends with
> 
> ```
> [Error - 3:04:08 PM] Connection to server got closed. Server will not be restarted.
> ```
> 
> The other day I used a GitHub Codespace to compile myself a vsix from the master branch. That worked fine and I can use for example the bash-completion support with that.

Same thing is happening for me too. Latest macOS and latest VS Code release.

**Comment by ehendrix23** [comment]
Just wanted to add, here as well.

**Comment by mojotx** [comment]
Me too

**Comment by skovhus** [comment]
This should be resolved in version 1.12.1, let me know if this work for you. And thanks for the quick feedback here. 👏

**Comment by ehendrix23** [comment]
Confirmed working
